packages/portal/src/components/search/SearchInterface.vue

```diff
@@ -533,6 +557,71 @@
 
       toggleFilterSheet() {
         this.$store.commit('search/setShowFiltersSheet', !this.$store.state.search.showFiltersSheet);
+      },
+
+      async addEntityValuesToAdvancedSearchFields(qfs, locale) {
+        // Filter the requested advanced search queries to those which need the entity look up
+        const qfsToLookUp = [].concat(qfs)
+          .map(query => {
+            return {
+              // remove the dash for fields with 'does not contain' modifier to include them in the qf's to look up
+              fieldWithoutModifier: query?.split(':')[0].replace('-', ''),
+              field: query?.split(':')[0],
+              value: query?.split(':')[1]
+            };
+          })
+          .filter(query => advancedSearchFieldsForEntityLookUp.map(field => field?.name).includes(query?.fieldWithoutModifier))
+          .map(query => {
+            return {
+              ...query,
+              suggestEntityType: advancedSearchFieldsForEntityLookUp.find(field => field?.name === query?.fieldWithoutModifier).suggestEntityType
+            };
+          });
+
+        if (qfsToLookUp.length) {
+          const fieldsWithEntityValues = [];
+
+          await Promise.all(
+            qfsToLookUp.map(async query => {
```

Comment:
Reduce the complexity and improve the readability of this by moving the function out to a new method, e.g. `await Promise.all(qfsToLookUp.map(this.lookupQfEntity))`

Reply:
Done. With further clean up and some renaming.